content/_guides/channelmodes.md

```diff
@@ -4,25 +4,224 @@ category: chanop
 credits: web7
 ---
 
-Various modes can be set on channels. Use `/mode #channel` to list current
-channel modes and `/msg chanserv info #channel` to list modes set with MLOCK.
+Channel modes and `ChanServ` flags are tools for configuring and managing
+channels.
 
-All channel modes will be lost when a channel becomes empty. Enable GUARD to
-preserve modes.
+Channel modes are the most basic form of channel configuration. Modes can be
+set using the IRC `/mode` command or, for some modes, via `ChanServ`.
 
-To set a mode, use `/mode #channel +(mode)` replacing `(mode)` with the letter
-that corresponds to the mode. To unset a mode, use `/mode #channel -(mode)`
+The `ChanServ` channel service provides additional administration functions
+for registered channels. It is complementary to modes. You can perform many of
+the same functions manually, but using `ChanServ` to administer your channel
+where applicable offers some benefits:
 
-## Available channel modes
+- It is always opped, so it can set and enforce IRC [channel
+  modes](#channel-modes) on your behalf.
+- It offers granular privileges through [user flags](#chanserv-user-flags),
+  allowing you to delegate limited capabilities to other users. 
+- It can maintain channel attributes and perform functions automatically
+  through [control flags](#chanserv-control-flags).
 
+## Channel modes
+
+Channel modes control:
+
+- Bans (`b`, `e`)
+- Moderation (`m`, `n`, `q`, `u`, `v`, `z`)
+- Forwarding (`f`, `F`)
+- Channel visibility (`p`, `s`)
+- Channel invitations (`g`, `i`, `I`)
+- Channel access controls (`j`, `k`, `l`, `n`, `o`, `Q`, `r`, `S`)
+
+### Requirements
+
+- To use `/mode`, you must be a channel operator (`+o`).
+- To use `ChanServ`, your channel must be registered and your privileges must
+  allow you to take actions that set modes.
+
+### Mode command summary
+
+| Purpose                    |Command                    |
+|----------------------------|---------------------------|
+| List current channel modes | `/mode <channel>`         |
+| Set a mode                 | `/mode <channel> +<mode>` |
+| Unset a mode               | `/mode <channel> -<mode>` |
+
+### Examples
+
+To allow only operators to set the topic on `#foo`:
+
+```irc
+/mode #foo +t
+```
+
+To set a user `bar` as an operator on `#foo` using the user's cloaked
+hostmask:
+
+```irc
+/mode #foo +o bar*!*@user/bar
```

Comment:
You can't do that in solanum, only `/mode #channel +o nick`

Reply:
Thank you. I conflated banning and opping in my mind. This would be correct, yes?

```
To ban any user connecting to #foo from a given host:

/mode #foo +b *!*@10-10-32-1.res.example.com
```

Reply:
yep!